programs/gateway/src/instructions/update_voter_weight_record.rs

```diff
@@ -55,3 +66,26 @@ pub fn update_voter_weight_record(
 
     Ok(())
 }
+
+
+/// Attempt to parse the account as a VoterWeightRecord or a TokenOwnerRecordV2
+/// depending on which one succeeds, return the voter weight.
+fn extract_input_voter_weight(input_account: &AccountInfo, registrar: &Registrar) -> u64 {
+    match registrar.predecessor_plugin_registrar {
+        None => {
+            // If there is no predecessor plugin registrar, then the input account must be a TokenOwnerRecordV2
+            let parse_result: core::result::Result<TokenOwnerRecordV2, ProgramError> = get_account_data(&registrar.governance_program_id, input_account);
+            match parse_result {
+                Ok(token_owner_record) => token_owner_record.governing_token_deposit_amount,
+                Err(_) => DEFAULT_VOTE_WEIGHT, // TODO should probably be an error
+            }
+        }
+        Some(predecessor) => {
+            let parse_result: core::result::Result<VoterWeightRecord, ProgramError> = get_account_data(&predecessor, input_account);
```

Comment:
We have to assert the deserialised `VoterWeightRecord` relationship with `Registrar.realm` and `Registrar.governing_token_mint`
nit: We can make it explicit and create a function `get_voter_weight_record_data_for_previous_plugin_and_registrar()`

Reply:
✅